x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -122,9 +124,11 @@ public boolean acquire(final EvictionListener listener) throws IOException {
                     listeners = Collections.unmodifiableSet(newListeners);
                     success = true;
                 } finally {
-                    if (success == false) {
-                        refCounter.decRef();
-                    }
+                    evictionLock.unlock();
```

Comment:
Before this change we released `evictionLock` after calling `refCounter.decRef()`, but this change reverses the order. Can we keep the order the same?

Reply:
Good catch - I've been a bit quick on this. I pushed 2180f565b935dca6ddbbcb55d2abb26a77ba857e